src/workerd/api/streams/standard.c++

```diff
@@ -1135,7 +1135,17 @@ WritableImpl<Self>::WritableImpl(
 template <typename Self>
 jsg::Promise<void> WritableImpl<Self>::abort(
     jsg::Lock& js, jsg::Ref<Self> self, v8::Local<v8::Value> reason) {
-  signal->triggerAbort(js, jsg::JsValue(reason));
+  // Per the spec, the signal.reason should be a DOMException with name 'AbortError'
+  // when no reason is provided, but the stored error should remain as the original reason.
+  auto signalReason = [&]() -> jsg::JsValue {
```

Comment:
Might require a compat flag.

Reply:
pedantic wpt then

Reply:
yep, `pedantic_wpt` should be just fine in this case.